overlay/overlay_exe/overlay_exe.cpp

```diff
@@ -0,0 +1,175 @@
+// run mumble_app.dll.
+static bool ConfigureEnvironment() {
+	std::wstring exe_path = GetExecutableDirPath();
+
+	// Remove the current directory from the DLL search path.
+	if (!SetDllDirectoryW(L""))
+		return false;
+
+	// Set mumble.exe's directory as the current working directory.
+	if (!SetCurrentDirectoryW(exe_path.c_str()))
+		return false;
+
+	return true;
+}
+
+// GetAbsoluteMumbleOverlayDllPath returns the absolute path to
+// mumble_ol.dll - the DLL containing the Mumble overlay code.
+static std::wstring GetAbsoluteMumbleOverlayDllPath() {
+	std::wstring exe_path = GetExecutableDirPath();
+	if (exe_path.empty())
+		return std::wstring();
+
+	std::wstring abs_dll_path(exe_path);
+	abs_dll_path.append(L"\\");
+	abs_dll_path.append(L"mumble_ol.dll");
+	return abs_dll_path;
+}
+
+int main(int argc, char **argv) {
+	UNUSED(argc);
+	UNUSED(argv);
+
+	// Mumble passes the OVERLAY_MAGIC_NUMBER it was built
+	// with the first argument to this helper process.
+	//
+	// This is done for two reasons.
+	//
+	// The first reason is that we need to ensure
+	// compatibility between Mumble, the overlay helper,
+	// and the overlay itself.
+	//
+	// The second reason is that it is an easy way to catch
+	// users accidentally double-clicking on the overlay
+	// helper's EXE file. We can detect these non-Mumble
+	// initiated launches by checking if we were passed any
+	// arguments at all. If no parameters are passed, we
+	// display a nice alert dialog directing users to
+	// 'mumble.exe' instead. 
+	unsigned int magic = 0;
+	{
+		std::wstring command_line(GetCommandLine());
+
+		// The command line will contain two consecutive spaces
+		// if the program was passed any arguments. If we don't
+		// find them, it probably means that a user has double-clicked
+		// the executable. Tell them to run 'mumble.exe' instead.
+		size_t sep = command_line.find(std::wstring(L"  "));
+		if (sep == std::string::npos) {
+			Alert(L"Mumble Overlay", L"This program is not meant to be run by itself. Run 'mumble.exe' instead.");
+			return -1;
+		}
+
+		// We expect that the Mumble process passes the overlay
+		// magic number that it is built with to us.
+		std::wstring magic_number_str = command_line.substr(sep);
+		try {
+			unsigned long passed_in_magic = std::stoul(magic_number_str);
+			magic = static_cast<unsigned int>(passed_in_magic);
+		} catch (std::exception &) {
+			return -2;
+		}
+	}
+
+	if (magic != OVERLAY_MAGIC_NUMBER) {
+		return -3;
+	}
+
+	if (!ConfigureEnvironment()) {
+		return -4;
+	}
+
+	std::wstring abs_dll_path = GetAbsoluteMumbleOverlayDllPath();
+	if (abs_dll_path.empty()) {
+		return -5;
+	}
+
+	HMODULE dll = LoadLibraryExW(abs_dll_path.c_str(), NULL, LOAD_WITH_ALTERED_SEARCH_PATH);
+	if (!dll) {
+		return -6;
+	}
+
+	OverlayHelperProcessMain entry_point = reinterpret_cast<OverlayHelperProcessMain>(GetProcAddress(dll, "OverlayHelperProcessMain"));
+	if (!entry_point) {
+		return -7;
```

Comment:
Log text to debugoutput? Sure we can check return codes; but can we instruct others to do the same? :)